org-code-javabuilder/theater/src/main/java/org/code/theater/CatImage.java

```diff
@@ -0,0 +1,80 @@
+package org.code.theater;
+
+import ij.IJ;
+import ij.ImagePlus;
+import ij.process.ImageProcessor;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import org.code.protocol.GlobalProtocol;
+import org.code.protocol.OutputAdapter;
+
+/**
+ * This is a sample library with some image manipulation that shows how theater could use the
+ * framework that has been set up. This is for demo purposes only. No production code should be
+ * added here and the class should be deleted before we ship Theater.
+ */
+public class CatImage {
+  private final OutputAdapter outputAdapter;
+
+  public CatImage() {
+    this.outputAdapter = GlobalProtocol.getInstance().getOutputAdapter();
+  }
+
+  public CatImage(OutputAdapter outputAdapter) {
+    this.outputAdapter = outputAdapter;
+  }
+
+  public void buildImageFilter() {
+    ImagePlus image = null;
+    try {
+      image =
+          IJ.openImage(
+              Paths.get(CatImage.class.getClassLoader().getResource("sampleImageBeach.jpg").toURI())
+                  .toString());
+    } catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+
+    // Uncommenting the following line and setting it to a local directory will save the image to
+    // your local computer
+    // IJ.save(image, "<my local directory>\beach.gif");
+
+    ImageProcessor ip = image.getProcessor();
+    BufferedImage bufferedImage = (BufferedImage) ip.createImage();
+    outputAdapter.sendMessage(ImageEncoder.encodeImageToMessage(bufferedImage));
+  }
+
+  public void buildCanvas() {
+    ImagePlus image = IJ.createImage("test", 300, 150, 1, 24);
```

Comment:
general q: are we thinking we will use ImageJ for image creation in the 'real' version or is this a placeholder?

Reply:
This is a placeholder. We certainly can use ImageJ, but it's not required.

Reply:
(I had used it while investigating how to set up Theater and figured I might as well include it for demo purposes in case it would be useful down the line)